python/cudf/cudf/core/udf/row_function.py

```diff
@@ -62,7 +66,8 @@ def _get_frame_row_type(dtype):
         fields.append((name, infos))
 
         # increment offset by itemsize plus one byte for validity
-        offset += elemdtype.itemsize + 1
+        itemsize = itemsizes.get(elemdtype) or elemdtype.itemsize
```

Comment:
You're reengineering `get` itself here a little 
```suggestion
        itemsize = itemsizes.get(elemdtype, elemdtype.itemsize)
```